### ISSUE TITLE
[Bug] CVE-2021-44228 Log4j

### ISSUE THREAD

**Issue by onetwopunch** [ray-project/ray#21055]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core

### What happened + What you expected to happen

It seems Ray is vulnerable to the latest critical Log4j vulnerability [CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228) see [this post](https://www.govcert.ch/blog/zero-day-exploit-targeting-popular-java-library-log4j/) for a detailed writeup.

### Versions / Dependencies

Any version of Ray that uses Log4j < 2.15

Found in our environment in `/opt/conda/lib/python3.7/site-packages/ray/jars/ray_dist.jar` We also found this in the `ray:latest` docker container at `sha256:055fd11dd70ac3f403b9356fb48d9dc37217656ae228d4fa395e4c8078e75975`

### Reproduction script

N/A

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by pcmoritz** [comment]
If you need a very short term fix right now, you can remove the `ray_dist.jar` from the wheel (it is not used by Ray unless you are using Ray Java bindings) like this (for example for macOS):

```
unzip ray-1.9.0-cp37-cp37m-macosx_10_15_intel.whl
rm ray-1.9.0.data/purelib/ray/jars/ray_dist.jar
zip -r ray-1.9.0-cp37-cp37m-macosx_10_15_intel.whl ray-1.9.0.data ray-1.9.0.dist-info
```

**Comment by pcmoritz** [comment]
Also cc @iycheng who is leading the 1.9.1 release cc @zhe-thoughts

**Comment by onetwopunch** [comment]
Thanks for the quick response and workaround! We're happy to wait till the 1.9.1 release. Please feel free to close this once the release has shipped :)

**Comment by jovany-wang** [comment]
Hi @architkulkarni , what's the progress of 1.9.1?

**Comment by architkulkarni** [comment]
A release candidate 1.9.1rc0 should be released later today, and 1.9.1 should be released by the end of the week.

**Comment by architkulkarni** [comment]
The release candidate 1.9.1rc0 has been released!  Please try it out with `pip install ray==1.9.1rc0` and let us know if you run into any issues.

**Comment by architkulkarni** [comment]
It turns out `log4j==2.15.0` is still vulnerable to [CVE-2021-45046](https://nvd.nist.gov/vuln/detail/CVE-2021-45046), and therefore so is `ray==1.9.1rc0`.  

The `log4j` dependency has been upgraded to `2.16.0` in master: https://github.com/ray-project/ray/pull/21081. This will be included in the final release of `ray==1.9.1`.

**Comment by qbphilip** [comment]
Hello @architkulkarni, thanks for your dedication to fix the vulnerabilities and keep our deployments safe!

There was a new vulnerability found in `log4j==2.16.0` and a fix is published in `2.17.0`, see:
* https://logging.apache.org/log4j/2.x/security.html
* https://thehackernews.com/2021/12/new-local-attack-vector-expands-attack.html
* https://thehackernews.com/2021/12/apache-issues-3rd-patch-to-fix-new-high.html

**Comment by architkulkarni** [comment]
Ray 1.9.1 has been released: https://github.com/ray-project/ray/releases/tag/ray-1.9.1
Ray 1.9.1 has `log4j==2.16.0`.

In the first week of January we plan to release Ray 1.9.2 which will have `log4j==2.17.0`, which will resolve the vulnerability mentioned by @qbphilip.
